pkg/registry/node/etcd/etcd.go

```diff
@@ -102,3 +103,30 @@ var _ = rest.Redirector(&REST{})
 func (r *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.RoundTripper, error) {
 	return node.ResourceLocation(r, r.connection, r.proxyTransport, ctx, id)
 }
+
+type HostLocator interface {
+	HostKubeletPort(pod *api.Pod, ctx api.Context) (int, error)
+}
+
+type NodeStoreHostLocator struct {
+	nodeStore *REST
+}
+
+func NewNodeStoreHostLocator(nodeStore *REST) HostLocator {
```

Comment:
Return a *NodeStoreHostLocator, document that it is a HostLocator.

Also somewhere do the `var(_ = HostLocator(&NodeStoreHostLocator{}))` compiler check.